src/test/java/spoon/test/variable/VariableTest.java

```diff
@@ -37,6 +44,64 @@
 
 public class VariableTest {
 
+    @Test
+    public void test() {
+        // contract: one can use setType without having a very hard generics type checking error
+
+        // to get this done, we use the refactoring below
+        CtClass c = Launcher.parseClass("class C{ void f() {Object i=null; }}");
+        List<CtLocalVariable> localVariables = c.getElements(new TypeFilter<>(CtLocalVariable.class));
+        System.out.println(localVariables);
+
+        // before: "CtLocalVariable<?> lv" could not typecheck, with a hard generics error
+        // now it works; we can use both "CtLocalVariable<?> lv" and "CtLocalVariable lv"
+        // it's source level backward compatible for Spoon itself
+        CtLocalVariable<?> lv = localVariables.get(0);
+        lv.setType(c.getFactory().Type().createReference(Object.class));
+
+        CtLocalVariable lv2 = localVariables.get(0);
+        lv2.setType(c.getFactory().Type().createReference(Object.class));
+
+    }
+
+    @Test
+    public void refactor() {
+        // rectoring all setType(CtTypeParameter<T>) -> setType(CtTypeParameter)
+        // this is the refactoring done for this pull-request
+        Launcher l = new Launcher();
+        l.addInputResource("src/main/java");
+        l.setSourceOutputDirectory("src/main/java");
+        l.getEnvironment().setPrettyPrinterCreator(() -> {
+                    return new SniperJavaPrettyPrinter(l.getEnvironment());
+                }
+        );
+        l.addProcessor(new AbstractProcessor<CtMethod<?>>() {
+            @Override
+            public boolean isToBeProcessed(CtMethod<?> candidate) {
+                if (!candidate.getSimpleName().equals("setType")) return false;
+                if (candidate.getParameters().size()!=1) return false;
+                final CtParameter<?> ctParameter = candidate.getParameters().get(0);
+                //System.out.println(ctParameter.getType().getSimpleName());
+                System.out.println(ctParameter.toString());
+                System.out.println(ctParameter.getType().getActualTypeArguments().size());
+                if (!ctParameter.getType().getSimpleName().equals("CtTypeReference")) return false;
+                if (ctParameter.getType().getActualTypeArguments().size()!=1) return false;
+                System.out.println(ctParameter.getType().toString());
+                return true;
+            }
+
+            @Override
+            public void process(CtMethod<?> element) {
+                System.out.println(element.getSignature());
```

Comment:
We can probably remove the prints now (here and in various places) :)

Reply:
done